### DISCUSSION TITLE
Computing dynamic correlation functions?

### DISCUSSION THREAD

**Discussion by jeohjee** [netket/netket#1492] [Q&A] [answered]
I am wondering how it would be possible to compute dynamic correlation functions, e.g. dynamic spin structure factor <S_z(k,t) S_z(-k,0)> in NetKet. 

According to a recent paper (https://arxiv.org/pdf/2303.08184.pdf), one should compute with TDVP the time-evolution of the state S_z | GS >, where | GS > is the ground state and can be obtained by using VMC with NetKet. NetKet also seems to nicely provide algorithm to perform TDVP. However, I am puzzled on how to initialize TDVP to act on S_z | GS >. How can I implement the S_z operation on the obtained |GS>? I am not expert on TDVP so I am not sure whether my question even makes too much sense.

Of course, if there is some alternative way to access dynamic correlators with NetKet, that would be wonderful!

Thanks for help in advance!

**Comment by gcarleo** [comment, marked as answer]
The operator Sz is diagonal in the computayional basis, so after you obtain a variational GS you need to define another model which is just  psi(z) =z1*GS(z1, z2,..), assuming you have Sz acting on spin 1. notice that variational models in netket are defined in log space, so you will need to remember that when defining this modified model

**Reply by jeohjee** [reply]
Thanks for the help! This method indeed implements the action of an operator to the ground state.
